pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/StoolUploadRoute.java

```diff
@@ -0,0 +1,182 @@
+
+import lombok.extern.log4j.Log4j2;
+import org.broadinstitute.dsm.db.dao.stoolupload.StoolUploadDao;
+import org.broadinstitute.dsm.db.dto.stoolupload.StoolUploadDto;
+import org.broadinstitute.dsm.exception.FileColumnMissing;
+import org.broadinstitute.dsm.exception.FileWrongSeparator;
+import org.broadinstitute.dsm.exception.UploadLineException;
+import org.broadinstitute.dsm.model.StoolUploadObject;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.statics.UserErrorMessages;
+import org.broadinstitute.dsm.util.SystemUtil;
+import org.broadinstitute.dsm.util.UserUtil;
+import org.broadinstitute.dsm.util.proxy.jackson.ObjectMapperSingleton;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+@Log4j2
+public class StoolUploadRoute extends RequestHandler {
+
+    private static final String PARTICIPANT_ID = "participantId";
+    private static final String MF_BARCODE = "mfBarcode";
+    private static final String RECEIVE_DATE = "receiveDate";
+    public static final String UPLOAD_ANYWAY = "uploadAnyway";
+
+
+    @Override
+    protected Object processRequest(Request request, Response response, String userId) throws Exception {
+        QueryParamsMap queryParams = request.queryMap();
+        String realm;
+
+        if (queryParams.value(RoutePath.REALM) != null) {
+            realm = queryParams.get(RoutePath.REALM).value();
+        } else {
+            throw new RuntimeException("No realm query param was sent");
+        }
+        String userIdRequest = UserUtil.getUserId(request);
+
+        if (UserUtil.checkUserAccess(realm, userId, "kit_upload", userIdRequest)) {
+            String content = request.body();
+
+            AtomicBoolean uploadAnyway = new AtomicBoolean(false);
+            if (queryParams.value(UPLOAD_ANYWAY) != null) {
+                uploadAnyway.set(queryParams.get(UPLOAD_ANYWAY).booleanValue());
+            }
+
+            try {
+                List<StoolUploadObject> stoolUploadContent;
+                if (uploadAnyway.get()) {
+                    stoolUploadContent =
+                            Collections.singletonList(ObjectMapperSingleton.instance().readValue(content, StoolUploadObject.class));
+                } else {
+                    try {
+                        stoolUploadContent = isFileValid(content);
+                    } catch (Exception e) {
+                        response.status(500);
+                        return e.getMessage();
+                    }
+                }
+
+                final List<StoolUploadObject> stoolUploadObjects = stoolUploadContent;
+                if (stoolUploadObjects.isEmpty()) {
+                    return "Text file was empty or couldn't be parsed to the agreed format";
+                }
+
+                stoolUploadObjects.forEach(stoolUploadObject -> {
+                    StoolUploadDao stoolUploadDao = new StoolUploadDao();
```

Comment:
Major: Let's avoid creating a new instance of StoolUploadDao for each iteration. We can extract StoolUploadDao as an instance field and inject it from the client or directly create new instance in the processRequest method.